### ISSUE TITLE
`no-useless-constructor` / `no-empty-function` with mixed JS & TS codebase

### ISSUE THREAD

**Issue by scottohara** [eslint/typescript-eslint-parser#418]
(As discussed in https://github.com/eslint/typescript-eslint-parser/issues/77#issuecomment-350872118)

**What version of TypeScript are you using?**
2.6.2

**What version of `typescript-eslint-parser` are you using?**
9.0.0

**What code were you trying to parse?**

```typescript
export default class Foo {
  constructor(private name: string) {}

  get greeting() : string {
    return `Hello ${this.name}`;
  }
}
```

**What did you expect to happen?**
No errors/warnings about empty constructor.

**What happened?**

```
  2:2   error  Useless constructor           no-useless-constructor
  2:36  error  Unexpected empty constructor  no-empty-function
```

For now, I have turned off these two rules in my `.ts` projects; but for projects that mix `*.ts` and `*.js` code, it would be nice to be able to have these rules enabled to catch any useless constructors/empty functions in `*.js` code without it also warning about valid `*.ts` constructors.

**Comment by j-f1** [comment]
This would probably have to be fixed by forking those rules into `eslint-plugin-typescript` and silencing​ warnings where they’re not appropriate.

**Comment by JamesHenry** [comment]
Yes, @j-f1 is correct! This will definitely have to be addressed via a new rule, because it is entirely TypeScript-specific.

@scottohara Please can you open an issue on https://github.com/nzakas/eslint-plugin-typescript?

**Comment by scottohara** [comment]
Sure, no problem.  https://github.com/nzakas/eslint-plugin-typescript/issues/101
